File: python/01-Fundamentals/03-fundamentals.py
#1 - Biggie Size
def biggie(arr):
    for i in range(len(arr)):
        if arr[i] > 0:
            arr[i] = 'big'
    return arr
#print(biggie([-1,3,5,-5]))

#2 - Previous Lengths
def prevLength(arr):
    for i in range(len(arr)-1,-1,-1):
        if i == 0:
            arr[i] = 0
            return arr
        arr[i] = len(arr[i-1])
    return arr
#print(prevLength(["I", "am", "the", "best"]))

#3 - Print Low, Return High
def minReturnMax(arr):
    min = arr[0]
    max = arr[0]
    for i in range(len(arr)):
        if arr[i] < min:
            min = arr[i]
        if arr[i] > max:
            max = arr[i]
    print(min)
    return max
#print(minReturnMax([2,7,3,5,8,9,3]))

#4 - Add Seven to Most
def addSeven(arr):
    newArray = []
    for i in range(1,len(arr), +1):
        newArray.append(arr[i]+7)
    return newArray
#print(addSeven([1,2,3,4]))

#5 - Print One, Return Another
def secondToLast(arr):
    print(arr[len(arr)-2])
    odd = 0
    for i in range(len(arr)):
        if arr[i]% 2 != 0:
            odd = arr[i]
            return odd
#print(secondToLast([6,3,8,9,1,2,5,8]))

#6 - Reverse Array
def reverseArray(arr):
    for i in range(len(arr)/2):
        temp = arr[i]
        arr[i] = arr[len(arr)-1-i]
        arr[len(arr)-1-i] = temp
    return arr
#print(reverseArray([3,1,6,4,2,9]))

#7 - Double Vision
def doubleTrouble(arr):
    newArray = []
    for i in range(len(arr)):
        newArray.append(arr[i]*2)
    return newArray
#print(doubleTrouble([1,2,3]))

#8 - Outlook: Negative
def negatives(arr):
    newArray = []
    for i in range(len(arr)):
        if arr[i] < 0:
            newArray.append(arr[i])
        else:
            newArray.append(arr[i]*-1)
    return newArray
#print(negatives([1,-3,5]))

#9 - Count Positives
def countPositives(arr):
    sum = 0
    for i in range(len(arr)):
        if arr[i] > 0:
            sum += 1
    arr[len(arr[i])] = sum
    return arr
#print(countPositives([-1,1,1,1]))

#10 - Always Hungry
def alwaysHungry(arr):
    count = 0
    for i in range(len(arr)):
        if arr[i] == "food":
            print("mmmm... yummy.")
            count += 1
    if count == 0:
        print("I'm hungry!")
#alwaysHungry(["word", False, "another string", 34, "no", "blah"])

#11 - Evens and Odds
def evensAndOdds(arr):
    for i in range(len(arr)-2):
        if arr[i] % 2 != 0 and arr[i+1] % 2 != 0 and arr[i+2] % 2 != 0:
            print("That's odd!")
        if arr[i] % 2 == 0 and arr[i+1] % 2 == 0 and arr[i+2] % 2 == 0:
            print("Even more so!")
#evensAndOdds([1,3,9,4,5,6,7,8,9,2])

#12 - Swap Toward the Center
def swapInwards(arr):
    for i in range(0,len(arr)/2, +2):
        temp = arr[i]
        arr[i] = arr[len(arr)-1-i]
        arr[len(arr)-1-i] = temp
    return arr
#print(swapInwards([1,2,3,4,5,6]))

#13 - Increment the Seconds
def incrementOdds(arr):
    for i in range(len(arr)):
        if i % 2 != 0:
            arr[i] += 1
    return arr
#print(incrementOdds([1,2,3,4,5,6]))

#14 - Scale the Array
def scaleable(arr, num):
    for i in range(len(arr)):
        arr[i] *= num
    return arr
print(scaleable([1,2,3,4,5,6], 2))

File: javaScript/01-Fundamentals/03-fundamentals.js
'use strict';
//1
function biggieSize(input){
    for(var i = 0; i < input.length; i++){
        if(input[i] > 0){
            input[i] = "big";
        }
    }
    return input;
}
//console.log(biggieSize([-1,3,5,-5]));

//2
function prevLen(input){
    for(var i = input.length-1; i >= 0 ; i--){
        if(i == 0){
            input[i] = 0;
            return input;
        }
        input[i] = input[i-1].length;
    }
    return input;
}
//console.log(prevLen(["boo3", "haha", "gotchya", "kiddo"]));

//3
function printLowReturnHigh(input){
    var min = input[0];
    var max = input[0];
    for(var i = 0; i < input.length; i++){
        if(input[i] < min)
            min = input[i];
        if(input[i] > max)
            max = input[i];
    }
    console.log(min);
    return max;
}
//console.log(printLowReturnHigh([-34,5,-5,24,14]));

//4
function addSeven(input){
    var newArray = [];
    for(var i = 1; i < input.length; i++){
        newArray.push(input[i]+7);
    }
    return newArray;
}
//console.log(addSeven([2,1,3,4]));

//5
function printAndReturn(input){
    console.log(input[input.length-2]);
    for(var i = 0; i < input.length; i++){
        if(input[i] % 2 != 0){
            return input[i];
        }
    }
}
//console.log(printAndReturn([1,14,3,4]));

//6
function reverseArray(input){
    if(input.length < 2){
        return input;
    }
    for(var i = 0; i < input.length/2; i++){
        var temp = input[i];
        input[i] = input[input.length-1-i];
        input[input.length-1-i] = temp;
    }
    return input;
}
//console.log(reverseArray([1,2,3,4,5,6]));

//7
function doubleVision(input){
    var newArray = [];
    for(var i = 0; i < input.length; i++){
        newArray.push(input[i]*2);
    }
    return newArray;
}
//console.log(doubleVision([2,3,4,5]));

//8
function negativeOutlook(input){
    var newArray = [];
    for(var i = 0; i < input.length; i++){
        if(input[i] > 0){
            newArray.push(input[i]*(-1));
        }
        else if(input[i] <= 0){
            newArray.push(input[i]);
        }
    }
    return newArray;
}
//console.log(negativeOutlook([12,3,-5]));

//9
function endPositive(arr){
    var sum = 0;
    for(var i = 0; i < arr.length; i++){
        if(arr[i] > 0){
            sum += 1;
        }
    }
    arr[arr.length-1] = sum;
    return arr;
}
//console.log(endPositive([-1,1,1,1]));

//10
function alwaysHungry(arr){
    var food = 0;
    for(var i = 0; i < arr.length; i++){
        if(arr[i] == "food"){
            console.log("yummy");
            food += 1;
        }
    }
    if(food == 0){
        console.log("I'm hungry");
    }
}
//alwaysHungry(["word", false, "food", 34, "no", "blah"]);

//11
function evenAndOdds(arr){
    for(var i = 0; i < arr.length-2; i++){
        if(arr[i] % 2 != 0 && arr[i+1] % 2 != 0 && arr[i+2] % 2 != 0){
            console.log("That’s odd!");
        }
        else if(arr[i] % 2 == 0 && arr[i+1] % 2 == 0 && arr[i+2] % 2 == 0){
            console.log("Even more so!");
        }
    }
}
//evenAndOdds([1,3,5,4,5,6,7,8,9,2]);

//12
function swapInwards(arr){
    for(var i = 0; i < arr.length/2; i+=2){
        var temp = arr[i];
        arr[i] = arr[arr.length-1-i];
        arr[arr.length-1-i] = temp;
    }
    return arr;
}
console.log(swapInwards([1,2,3,4,5,6]));

//13
function incrementTheSeconds(arr){
    for(var i = 0; i < arr.length; i++){
        if(arr[i] % 2 != 0){
            arr[i] += 1;
            console.log(arr[i]);
        }
        console.log(arr[i]);
    }
    return arr;
}
//console.log(incrementTheSeconds([1,2,3,4,5,6]));

//14
function scaleTheArray(arr, num){
    for(var i = 0; i < arr.length; i++){
        arr[i] *= num;
    }
    return arr;
}
//console.log(scaleTheArray([1,2,3,4,5,6], 2));